V2/api/evmWithdrawPrehooktoEVM/src/index.ts

```diff
@@ -0,0 +1,151 @@
+
+// Define Aave pool ABI (just the withdraw function)
+const aavePoolAbi = [
+  "function withdraw(address asset, uint256 amount, address to) external returns (uint256)"
+];
+
+// Creating Contract interfaces
+const withdrawTokenContract = new ethers.Contract(withdrawToken, erc20Abi, signer);
+const aavePoolContract = new ethers.Contract(aavePoolAddress, aavePoolAbi, signer);
+
+// Function to get the optimal route for the swap using Squid API
+const getRoute = async (params: any) => {
+  try {
+    const result = await axios.post(
+      "https://apiplus.squidrouter.com/v2/route",
+      params,
+      {
+        headers: {
+          "x-integrator-id": integratorId,
+          "Content-Type": "application/json",
+        },
+      }
+    );
+    const requestId = result.headers["x-request-id"];
+    return { data: result.data, requestId: requestId };
+  } catch (error) {
+    if (error.response) {
+      console.error("API error:", error.response.data);
+    }
+    console.error("Error with parameters:", params);
+    throw error;
+  }
+};
+
+// Function to approve the transactionRequest.target to spend fromAmount of withdrawToken
+const approveSpending = async (transactionRequestTarget: string, tokenAmount: string) => {
+  try {
+    const currentAllowance = await withdrawTokenContract.allowance(signer.address, transactionRequestTarget);
+    if (currentAllowance.lt(tokenAmount)) {
+      const tx = await withdrawTokenContract.approve(transactionRequestTarget, tokenAmount);
+      await tx.wait();
+      console.log(`Approved ${tokenAmount} tokens for ${transactionRequestTarget}`);
+    } else {
+      console.log("Sufficient allowance already exists.");
+    }
+  } catch (error) {
+    console.error('Approval failed:', error);
+    throw error;
+  }
+};
+
+// Set up parameters for withdrawing from Aave and bridging to Optimism
+(async () => {
+  
+  // Construct the withdraw function call data using the contract interface
+  const withdrawData = aavePoolContract.interface.encodeFunctionData("withdraw", [
+    fromToken,
+    amount,
+    "0xEa749Fd6bA492dbc14c24FE8A3d08769229b896c" // Squid multi-router address
+  ]);
+
+  const params = {
+    fromAddress: signer.address,
+    fromChain: fromChainId,
+    fromToken: fromToken,
+    fromAmount: amount.toString(),
+    toChain: toChainId,
+    toToken: toToken,
+    toAddress: signer.address,
+    slippage: 1,
+    preHook: {
+      chainType: "evm",
+      fundAmount: amount.toString(),
+      fundToken: withdrawToken,
+      calls: [
+        {
+          callType: 2,
+          target: aavePoolAddress,
+          value: "0",
+          callData: withdrawData,
+          payload: {
+            tokenAddress: fromToken,
+            inputPos: 1,
+          },
+          estimatedGas: "150000",
+          chainType: "evm",
+        },
+      ],
+      provider: "Test",
+      description: "Test Arbitrum preHook",
+      logoURI: "https://valoraapp.com/favicon.ico",

```

Comment:
This might not be the right icon? Maybe use the Squid icon instead?

Reply:
Actually I see the other example uses this https://app.aave.com/favicon.ico
Could also update the description and provider

Reply:
Good call out, updated the provider, description, and logoURI